lib/Sema/CSSolver.cpp

```diff
@@ -1964,3 +1965,41 @@ bool DisjunctionChoice::isSymmetricOperator() const {
       paramList->get(1)->getInterfaceType()->getWithoutSpecifierType();
   return firstType->isEqual(secondType);
 }
+
+void DisjunctionChoice::propagateConversionInfo() const {
+  if (!CS->isExplicitConversionConstraint(Disjunction))
+    return;
+
+  auto LHS = Choice->getFirstType();
+  auto typeVar = LHS->getAs<TypeVariableType>();
+  if (!typeVar || typeVar->getImpl().hasRepresentativeOrFixed())
```

Comment:
I don't think it make sense to check `hasRepresentativeOrFixed` here.

Reply:
Ah right, because it has to be the same type variable in the choice, I'll remove it.

Reply:
Actually I remember now why it's required because of BindParam contraction e.g. `.map { $0 as AnyObject }`, we can't bind the argument directly.

Reply:
In that case wouldn't it make more sense to bail based on that constraint rather than whether there is a representative (since there are lots of cases where we've made multiple type variables part of the same equivalence class)?

Either way a comment here would be helpful.

Reply:
Sorry I don't really follow, there are not going to be any constraints left between param/arg in there (because contraction removes everything), all remaining constraints are going to be attached to argument type variable but `getPotentialBindings` is going to look at param type variable and fail because type variable associated to the argument was bound.

Reply:
So then should this be getting the representative if this is a type variable, and checking if *that* is bound, and if not, binding it?

Reply:
Ah, I understand what you mean now, it makes sense, let me try that... (why didn't it occur to me :))